File: lint-rules/src/main/java/MyIssueRegistry.java
import com.android.tools.lint.detector.api.Issue;
import com.example.lintrules.MyAnnotationDetector;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

public class MyIssueRegistry extends com.android.tools.lint.client.api.IssueRegistry {
    public MyIssueRegistry() {

    }


    @NotNull
    @Override
    public List<Issue> getIssues() {
        List<Issue> issues = new ArrayList<Issue>();
        issues.add(MyAnnotationDetector.CAREFUL_NOW_ISSUE);

        return issues;
    }
}

File: lintlib/src/main/java/MyIssueRegistry.kt
import com.android.tools.lint.detector.api.Category.Companion.CORRECTNESS
import com.android.tools.lint.detector.api.Implementation
import com.android.tools.lint.detector.api.Issue
import com.android.tools.lint.detector.api.Scope
import com.android.tools.lint.detector.api.Severity
import java.util.*


val CarefulNow = Issue.create("com.example.lintlib.CarefulNow11",
        "com.example.lintlib.CarefulNow CarefulNowCarefulNowCarefulNowCarefulNowCarefulNow",
        "Dont use! ",
        CORRECTNESS,
        5,
        Severity.WARNING,
        Implementation(MyDetector::class.java, Scope.JAVA_FILE_SCOPE))

class MyIssueRegistry : com.android.tools.lint.client.api.IssueRegistry() {
    override val issues: List<Issue>
        get() = listOf(CarefulNow)

    override val api: Int = com.android.tools.lint.detector.api.CURRENT_API
}
